core/src/main/java/org/mule/runtime/core/internal/resolving/DefaultExtensionResolvingContext.java

```diff
@@ -95,8 +101,16 @@ public BaseTypeBuilder getTypeBuilder() {
    */
   @Override
   public void dispose() {
-    if (connectionHandler != null) {
-      connectionHandler.release();
+    try {
+      connectionHandler.get().ifPresent(ConnectionHandler::release);
```

Comment:
if you don't require a connection, you here craete it just in order to release it. INstead of a supplier you should use a LAzyValue and use the ifComputed() function